parser-typechecker/src/Unison/PrintError.hs

```diff
@@ -609,6 +609,21 @@ renderTypeError e env src curPath = case e of
     Pr.hang
       "This case would be ignored because it's already covered by the preceding case(s):"
       (annotatedAsErrorSite src loc)
+  UnknownTerm {..}
+    | Var.typeOf unknownTermV == Var.MissingResult ->
+        mconcat
+          [ "The last statement of a block must be an expression, ",
+            "but this is a definition:\n\n",
+            annotatedAsErrorSite src termSite,
+            "\nI don't know what the result of this block should be.\n",
+            "Did you forget to add an expression at the end of the block?\n",
+            case expectedType of
+              Type.Var' (TypeVar.Existential {}) -> "I don't know what type it should be either."
+              _ ->
+                "It should be of type "
+                  <> style Type1 (renderType' env expectedType)
+                  <> "."
+          ]
```

Comment:
Suggestion to use `Pr.lines` and/or `Pr.wrap` as in the other cases. Don't use hard line breaks except when breaking paragraphs. This deals with terminal being too short for "The last element of a block must be an expression, but this is a definition."

Also simplifying language a bit, wdyt? 

```suggestion
    Pr.lines [ 
      Pr.wrap "The last element of a block must be an expression, but this is a definition:", 
      "",
      annotatedAsErrorSite src termSite,
      Pr.wrap $ "Try adding an expression at the end of the block." <> msg
      ]
      where
        msg = case expectedType of
          Type.Var' (TypeVar.Existential {}) -> mempty
          _ -> Pr.wrap $ "It should be of type " <> Pr.group (style Type1 (renderType' env expectedType) <> ".")
```

Reply:
My take is that "I don't know what type it should be either" sounds weird / somewhat confusing, especially for someone who doesn't already know that there's an alternate form of the message. It also doesn't really add any information. If there's no constraints, just don't tell the user about any constraints.